content/community/changelog/app-nuget/v5/breaking-changes/_index.en.md

```diff
@@ -0,0 +1,107 @@
+            {
+                // Your code here
+                return await Task.FromResult(layoutSettings);
+            }
+        }
+        ```
+
+## 2. Obsolete method GetOptionId removed from App/AppBase/IAltinnApp
+In [version 4.24.0](../../v4/whats-new/_index.en.md) we introduced a new way of supporting dynamic options making the GetOptionId methods in obsolete. The methods have now been removed and you should use the new way of implementing options as described [in the documentation](../../../../../app/development/data/options/_index.en.md)
+
+## 3. Obsolete method RunAppEvent removed from App/AppBase/IAltinnApp
+The RunAppEvent method is a old construct for hooking into various application events. This have been made obsolete by having concrete method overrides for each type of event as [described in the documentation](../../../../../technology/architecture/components/application/construction/altinn-apps/app/app-backend/applogic-events/_index.md). The RunAppEvent method was passed in an `AppEventType` enum which specified the type of event that was fired. You would then need to have code checking the type and performing the logic needed. The table below shows the old enum values and their corresponding new methods that should be used instead.
+
+| Enum                      |Corresponding method                       |
+| ---                       | ---                                       |
+| Calculation               | RunProcessDataRead or RunProcessDataWrite |
+| Instantiation             | RunDataCreation                           |
+| ValidateInstantiation     | RunInstantiationValidation                |
+| Validation                | RunDataValidation                         |
+| DataRetrieval             | RunProcessDataRead                        |
+| BeforeProcessChange       | OnStartProcess                            |
+| AfterProcessChange        | OnEndProcess                              |
+| AppModelCreation          | CreateNewAppModel                         |
+
+## 4. Obsolete methods RunCalculation removed from AppBase/IAltinnApp
+[In version 4.7.0](../../../../../community/changelog/app-nuget/_index.en.md) the RunCalculation method was replaced with the methods RunProcessDataRead and RunProcessDataWrite. RunCalculation has now been removed and those that have code in this method needs to move this to either RunProcessDataRead or RunProcessDataWrite.
+
+The process to update is:
+
+1. Add the DataProcessing folder and DataProcessingHandler class from our [app template](https://github.com/Altinn/altinn-studio/tree/master/src/Altinn.Apps/AppTemplates/AspNet/App/logic) to your app.
+2. Update App.cs. Add a class field for DataProcessingHandler and copy new methods (RunProcessDataRead and RunProcessDataWrite) from [App.cs](https://github.com/Altinn/altinn-studio/blob/master/src/Altinn.Apps/AppTemplates/AspNet/App/logic/App.cs)
```

Comment:
Wrong links, will fix. Good point on commit id.

Reply:
Hm, on this particular link it might be that master is best since we want them to use the latest.

Reply:
Yeah, might be fine in this case. This is only relevant when migrating from 4 to 5, so if it changes in the far future (f.ex from v 5 to 6), this may be irrelevant anyway.